python/pyspark/sql/functions.py

```diff
@@ -322,48 +354,80 @@ def signum(col):
     return _invoke_function_over_column("signum", col)
 
 
-@since(1.4)
 def sin(col):
     """
-    :param col: angle in radians
-    :return: sine of the angle, as if computed by `java.lang.Math.sin()`
+    .. versionadded:: 1.4.0
+
+    Parameters
+    ----------
+    col : :class:`Column` or str
+
+    Returns
+    -------
+    :class:`Column`
+        sine of the angle, as if computed by `java.lang.Math.sin()`
     """
     return _invoke_function_over_column("sin", col)
 
 
-@since(1.4)
 def sinh(col):
     """
-    :param col: hyperbolic angle
-    :return: hyperbolic sine of the given value,
-             as if computed by `java.lang.Math.sinh()`
+    .. versionadded:: 1.4.0
+
+    Parameters
+    ----------
+    col : :class:`Column` or str
+        hyperbolic angle
+
+    Returns
+    -------
+    :class:`Column`
+        hyperbolic sine of the given value,
+        as if computed by `java.lang.Math.sinh()`
     """
     return _invoke_function_over_column("sinh", col)
 
 
-@since(1.4)
 def tan(col):
     """
-    :param col: angle in radians
-    :return: tangent of the given value, as if computed by `java.lang.Math.tan()`
+    .. versionadded:: 1.4.0
+
+    Parameters
+    ----------
+    col : :class:`Column` or str
+        angle in radians
+
+    Returns
+    -------
+    :class:`Column`
+        tangent of the given value, as if computed by `java.lang.Math.tan()`
     """
     return _invoke_function_over_column("tan", col)
 
 
-@since(1.4)
 def tanh(col):
     """
-    :param col: hyperbolic angle
-    :return: hyperbolic tangent of the given value
-             as if computed by `java.lang.Math.tanh()`
+    .. versionadded:: 1.4.0
+
+    Parameters
+    ----------
+    col : :class:`Column` or str
+        hyperbolic angle
+
+    Returns
+    -------
+    :class:`Column`
+        hyperbolic tangent of the given value
+        as if computed by `java.lang.Math.tanh()`
     """
     return _invoke_function_over_column("tanh", col)
 
 
 @since(1.4)
```

Comment:
Not to change to `.. versionadded::`?

Reply:
Oh yeah. This one no need to change because `.. deprecated::` is still [reST style directive](https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html). So, this has no problem about formatting error because it does not use NumPy documentation specific syntax.